rustls/tests/api.rs

```diff
@@ -3321,9 +3321,9 @@ mod test_quic {
             .unwrap();
         assert_eq!(
             server.process_new_packets().err(),
-            Some(Error::PeerMisbehavedError(
-                "QUIC transport parameters not found".into(),
-            )),
+            Some(Error::PeerMisbehaved(
+                rustls::PeerMisbehaved::MissingQuicTransportParameters
```

Comment:
Style nit: maybe just import `PeerMisbehaved` in this module, as we do elsewhere?